packages/angular_devkit/build_angular/src/utils/purge-cache.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { BuilderContext } from '@angular-devkit/architect';
+import { PathLike, existsSync, promises as fsPromises } from 'fs';
+import { join } from 'path';
+import { normalizeCacheOptions } from './normalize-cache';
+
+/** Delete stale cache directories used by previous versions of build-angular. */
+export async function purgeStaleBuildCache(context: BuilderContext): Promise<void> {
+  const projectName = context.target?.project;
+  if (!projectName) {
+    return;
+  }
+
+  const metadata = await context.getProjectMetadata(projectName);
+  const { basePath, path, enabled } = normalizeCacheOptions(metadata, context.workspaceRoot);
+
+  if (!enabled || !existsSync(basePath)) {
+    return;
+  }
+
+  const entries = await fsPromises.readdir(basePath);
+
+  for (const entry of entries) {
+    const entryPath = join(basePath, entry);
+    if (entryPath === path) {
+      // Skip this directory since this is not stale.
+      continue;
+    }
+
+    // The below should be removed and replaced with just `rm` when support for Node.Js 12 is removed.
+    const { rm, rmdir } = fsPromises as typeof fsPromises & {
+      rm?: (
+        path: PathLike,
+        options?: {
+          force?: boolean;
+          maxRetries?: number;
+          recursive?: boolean;
+          retryDelay?: number;
+        },
+      ) => Promise<void>;
+    };
+
+    try {
+      if (rm) {
+        await rm(entryPath, { force: true, recursive: true, maxRetries: 3 });
+      } else {
+        await rmdir(entryPath, { recursive: true, maxRetries: 3 });
+      }
+    } catch {}
```

Comment:
I feel like I just saw this snippet in another PR, is this a common pattern in the CLI? Would it be worth factoring this out to something reusable?

Reply:
Yeah, I can do that once both PRs are merged.

Reply:
Actually the other PR is using the sync version.

Reply:
Up to you whether it's worth merging them, just something I noticed looking through it.